shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/PParams.hs

```diff
@@ -232,3 +262,155 @@ emptyPParams =
      , _extraEntropy = NeutralNonce
      , _protocolVersion = ProtVer 0 0
      }
+
+-- | Update Proposal
+data Update crypto
+  = Update (PPUpdate crypto) EpochNo
+  deriving (Show, Eq, Generic)
+
+instance NoUnexpectedThunks (Update crypto)
+
+instance Crypto crypto => ToCBOR (Update crypto) where
+  toCBOR (Update ppUpdate e) =
+    encodeListLen 2 <> toCBOR ppUpdate <> toCBOR e
+
+instance Crypto crypto => FromCBOR (Update crypto) where
+  fromCBOR =
+    Update <$ enforceSize "Update" 2
+      <*> fromCBOR
+      <*> fromCBOR
+
+data PPUpdateEnv crypto = PPUpdateEnv SlotNo (GenDelegs crypto)
+  deriving (Show, Eq, Generic)
+
+instance NoUnexpectedThunks (PPUpdateEnv crypto)
+
+type PParamsUpdate = PParams' Maybe
```

Comment:
[bikeshedding] It's somewhat confusing to have both `PPUpdate` and `PParamsUpdate`! Maybe rename `PPUpdate` to something like `ProposedPPUpdates`?

Reply:
that's completely fair! I will change `PPUpdate` to `ProposedPPUpdates`, thanks.